_posts/2019/2019-05-12-watch-out-for-nils-in-ranges.md

```diff
@@ -0,0 +1,49 @@
+---
+title: Watch Out For nil in Ranges
+description: To Infinity and Beyond
+layout: article
+category: ruby
+image:
+  base: '2019/watch-out-for-nil-in-ranges'
+  alt: 'Buzz Lightyear'
+---
+
+Ruby 2.6 enhanced the syntax for ranges to allow for infinite values at either end. The new syntax is described in the [Ruby documentation for `Range`](http://ruby-doc.org/core-2.6.3/Range.html#class-Range-label-Endless+Ranges).
+
+This is great for representing concepts that require that, for example date ranges between a date and into the future, but the new “sugared” syntax changes existing behaviour.
+
+
+## Instead of…
+
+…raising an `ArgumentError` when a `nil` is passed into a range…
```

Comment:
I'm confused. I'm thinking: Why is `ArgumentError` being discussed here? I don't think the intro has set me to understand what this example is going to show.